core/src/main/scala/com/microsoft/azure/synapse/ml/codegen/Wrappable.scala

```diff
@@ -419,42 +419,48 @@ trait RWrappable extends BaseWrappable {
     thisStage.params.map { p =>
       val value = getParamInfo(p).rTypeConverter.map(tc => s"$tc(${p.name})").getOrElse(p.name)
       p match {
-        case p: ServiceParam[_] =>
-          s"""invoke("set${p.name.capitalize}Col", ${p.name}Col) %>%\ninvoke("set${p.name.capitalize}", $value)"""
-        case p =>
-          s"""invoke("set${p.name.capitalize}", $value)"""
+        case sp: ServiceParam[_] =>
+          val colName = sp.name + "Col"
+          getRConditionalSetterLine(colName, colName) + "\n" + getRConditionalSetterLine(sp.name, value)
+        case _: TypedIntArrayParam | _: TypedDoubleArrayParam =>
+          getRConditionalSetterLine(p.name, value)
+        case _: TypedArrayParam[_] | _: ArrayMapParam | _: JsonEncodableParam[_] =>
+          getRConditionalSetterLine(p.name, value, setterSuffix = "R")
+        case _ =>
+          getRConditionalSetterLine(p.name, value)
       }
-    }.mkString(" %>%\n")
+    }.mkString("\n")
+  }
+
+  private def getRConditionalSetterLine(name: String, value: String, setterSuffix: String = ""): String = {
+      s"""if (!is.null(${name})) mod <- mod %>% """ +
+        s"""invoke("set${name.capitalize}${setterSuffix}", $value)"""
```

Comment:
is there any way this logic can be added to the params themselves so we can avoid "exceptions" here

Reply:
So we would need to figure out a way not to use the casts that are inserted by DefaultParamInfo.  For instance, passing null to as.array results in an exception. Passing null to as.integer or as.double results in a value of 0, which might not be what you want.    I did try converting all the R casts (like as.double in DoubleInfo) to None, but that didn't compile.  I also tried removing all the casts from the generated src wrappers, and the tests all passed, so the casts are not necessary. If you can advise on how to fix DefaultParamInfo, I can give it a try.

Reply:
Here's the deal with the null checks we talked about on Thursday. Many arguments to the R wrappers default to null. I had said earlier that removing the casts resulted in full test passage, but further experiments showed that to be false.

o If we go with no null checks and no argument casts, we have a nice, elegant pipeline for the call, but it turns out that nulls passed to arguments that are typed as Integer or Double cause exceptions in Scala, since variables of those types can't be null.
o If we go with no null checks but opt for argument casts, we get exceptions in R. as.array(NULL) yields an exception in call cases.  as.integer(NULL) and as.double(NULL) return 0 if performed in isolation, but they get exceptions when used as an argument to be serialized.
o With both null checks and argument casts we pass all tests, locally at least.